src/schemaType/DocumentArrayType.ts

```diff
@@ -60,21 +60,26 @@ class DocumentArrayType extends BaseSchemaType {
 	}
 
 	/** Validate the document array */
-	public async validate(documentList: Document[]): Promise<string[]> {
+	public async validate(documentList: Document[]): Promise<Map<string, string[]>> {
 		// combining all the validation into one array of promise.all
 		// - validation against the documents in the array will return a single object with 0 to n keys - only those with keys indicate errors;
-		// - iterate the returned object and return the messages from each
-		// - flatten the final results
-
-		return (
-			await Promise.all(
-				documentList.map(async (document) => {
-					const documentErrors = await document.validate();
-
-					return Array.from(documentErrors.values()).map((err) => ensureArray(err));
-				}),
-			)
-		).flat(2);
+		const errorsMap = new Map<string, string[]>();
+		await Promise.all(
+			documentList.map(async (document, index) => {
+				const documentErrors = await document.validate();
+
+				const indexString = String(index);
+				documentErrors.forEach((messages, key) => {
+					const errors = errorsMap.get(key) ?? [];
```

Comment:
This doesn't seem correct. The `errorsMap` is keyed by the combined keypath (as seen in the `set` call below), but this is getting by the keypath returned from the validation routine.